Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,141 @@
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackageProduct(url: .init(finalBinPath))
+            alreadyExisting.append(pkgInstance)
+            
+            try JSONEncoder().encode(path: installedPackagesJSONPath, fileSystem: tool.fileSystem, alreadyExisting)
+        }
+    }
+    
+    struct Uninstall: SwiftCommand {
+        
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(commandName: "experimental-uninstall", abstract: "Offers the ability to uninstall executable products of installed package products")
+        }
+        
+        @OptionGroup
+        var globalOptions: GlobalOptions
+        
+        @Argument(help: "Name of the executable to remove.")
+        var name: String
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.dotSwiftPM
+            var alreadyRegistered = (try? InstalledPackageProduct.registered(tool.fileSystem)) ?? []
+            
+            guard let whatWeWantToRemove = alreadyRegistered.first(where: { $0.name == name || $0.url.basename == name }) else {
+                // The installed executable doesn't exist - let the user know, and stop here.
+                var stringErrorToShowTheUser = "No such installed executable as \(name)"
+                
+                // and, in case there are any installed executables, let the user know which ones do exist.
+                if !alreadyRegistered.isEmpty {
+                    stringErrorToShowTheUser += ", existing installed executables: \(alreadyRegistered.map(\.name).joined(separator: "\n"))"
+                }
+                
+                throw StringError(stringErrorToShowTheUser)
+            }
+            
+            try tool.fileSystem.removeFileTree(whatWeWantToRemove.url)
+            print("Removed \(name).")
+        }
+    }
+}
+
+fileprivate struct InstalledPackageProduct: Codable, Equatable {
+    
+    static func registered(_ fileSystem: FileSystem) throws -> [InstalledPackageProduct] {
+        let binPath = try fileSystem.getOrCreateSwiftPMInstalledBinariesDirectory()
+        
+        let contents = ((try? fileSystem.getDirectoryContents(binPath)) ?? [])
+            .map { binPath.appending($0) }
+        
+        return contents.map { path in
+            InstalledPackageProduct(url: .init(path))
+        }
+    }
+    
+    /// The name of this installed product, being the basename of the URL.
+    var name: String {
+        url.basename
+    }
+    
+    /// Path of the executable
+    let url: AbsolutePath
```

Comment:
Please use descriptive property names, as this isn't a URL, but an actual filesystem path.

```suggestion
    let path: AbsolutePath
```